user_payload_provider/mock/provider.go

```diff
@@ -0,0 +1,18 @@
+package mock
```

Comment:
I'm not sure, but exported methods should have comments. Maybe that can be omitted for mock packages.